### PR TITLE
feat:                   units follow-up with testing

### PR DESCRIPTION
**Pull request by seanpearsonuk**
- **Follow-up on initial units integration into PyFluent.**
- upgrade ansys-units dependency
- enable a temporarily disabled test step due to this upgrade
- extend the testing
- fix an flobject assignment issue where exceptions were inadvertently hidden
- treat None value for units-quantity to indicate dimensionless
- implement exception handling
- rename `get_state_as_quantity()` (too long) as `as_quantity()`
- make the generation of the table data stricter

### CODE REVIEW

File: src/ansys/fluent/core/solver/flunits.py [hunk 1]
```diff
@@ -43,9 +93,7 @@ def make_python_fl_unit_table(scheme_unit_table):
     "collision-rate": "m^-3 s^-1",
     "concentration": "kmol m^-3",
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/solver/flunits.py:94]: mol, not kmol is an SI unit so this is wrong

File: src/ansys/fluent/core/solver/flunits.py [hunk 2]
```diff
@@ -109,14 +146,9 @@ def make_python_fl_unit_table(scheme_unit_table):
     "pressure-time-deriv-sqr": "Pa^2 s^-2",
     "pressure-time-derivative": "Pa s^-1",
     "resistance": "m^-1",
-    "site-density": "kgmol m^-2",
     "soot-formation-constant-unit": "kg N^-1 m^-1 s^-1",
-    "soot-limiting-nuclei-rate": "1e+15-particles/m3-s",
     "soot-linear-termination": "m^3 s^-1",
-    "soot-oxidation-constant": "kg m kgmol^-1 K^-0.5 s^-1",
-    "soot-pre-exponential-constant": "1.e15 kg^-1 s^-1",
     "soot-sitespecies-concentration": "kmol m^-3",
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/solver/flunits.py:151]: mol not kmol is an SI unit so this is wrong

File: src/ansys/fluent/core/solver/flunits.py [hunk 3]
```diff
@@ -59,13 +107,9 @@ def make_python_fl_unit_table(scheme_unit_table):
     "elec-charge": "A h",
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/solver/flunits.py:107]: this is wrong. 
I'm going to add an extra constraint: Instantiate a unit quantity for each unit and compare its value to the SI value.

Comment (seanpearsonuk) [reply src/ansys/fluent/core/solver/flunits.py]: This is all done and updated

**Review by prmukherj — APPROVED** [review]

**Review by hpohekar — APPROVED** [review]

**Pull request merged**